stumpy/core.py

```diff
@@ -1774,7 +1903,7 @@ def preprocess_non_normalized(T, m):
     check_window_size(m, max_size=T.shape[-1])
     T_subseq_isfinite = rolling_isfinite(T, m)
     T[~np.isfinite(T)] = 0.0
-    T_subseq_isconstant = rolling_isconstant(T, m)
+    T_subseq_isconstant = rolling_isconstant(T, m)  # POTENTIALLY: INCORRECT / USELESS
```

Comment:
FYI: I added the comment in line above to remind myself regarding the following items. I will remove the comment.

1. I think computing `T_subseq_isconstant` after cleaning `T` can give wrong result.
2. I think we do not even need this line. This function regards the non-normalized preprocess, and, hence, I think there is no use for `T_subseq_isconstant`.

Reply:
> I think computing T_subseq_isconstant after cleaning T can give wrong result.

Is that because we should be assessing "constant" on the original unmodified `T` rather than assessing "constant" after values of `T` that are `np.inf` have been converted to `0.0`? For example:

```
T = [1.0, 0.0, 0.0, np.inf, 0.0., 0.0]
m = 3
```
In this case, when `m = 3`, none of the subsequences should be constant. However, if `np.inf` gets turned into `0.0` and then we assess "constant" then the last three subsequences end up being "incorrectly" constant. Is that what you mean?

> I think we do not even need this line. This function regards the non-normalized preprocess, and, hence, I think there is no use for T_subseq_isconstant.

Yeah, I noticed it too. I don't recall why it was added to non-normalized (maybe a mistake). Feel free to remove it if it is completely useless for non-normalized methods.

Note that it is being used in `core.preprocess_diagonal`, which likely needs `T_subseq_isconstant` so we'll need to add it there.

Reply:
Now that I think about it, I think we (I) got lazy and skipped the typical step of :

```
T[np.isinf(T)] = np.nan
```

However, I think it is safer for this function to be:

```
    T = _preprocess(T)
    check_window_size(m, max_size=T.shape[-1])
    T_subseq_isfinite = rolling_isfinite(T, m)
    T[~np.isfinite(T)] = np.nan
    T_subseq_isconstant = rolling_isconstant(T, m)
    T[np.isnan(T)] = 0
```

Even if we remove `rolling_isconstant`, let's split the `T[~np.isfinite(T)]` step into 2 separate steps as shown above to protect/remind ourselves in the future.

Reply:
> Is that what you mean?

Yes. 

> Feel free to remove it if it is completely useless for non-normalized methods.
> 
> Note that it is being used in `core.preprocess_diagonal`, which likely needs `T_subseq_isconstant` so we'll need to add it there.

Got it. I will do it.

Reply:
> let's split the `T[~np.isfinite(T)]` step into 2 separate steps as shown above to protect/remind ourselves in the future. 

Got it. Now that you have mentioned it, I am thinking that we may add the step `T[~np.isfinite(T)] = np.nan` in the function `core._preprocess`. Because, I think we always want to replace `np.inf` with `np.nan`. So, we can just do it in `_preprocess`.

What do you think?

Reply:
@NimaSarajpoor I get your point. However, I think `core._preprocess` as being much more primitive/simple than `core.process` in that `core._preprocess` is operating at a "higher" "array type" level (i.e., checking if it is dataframe, converting it to a numpy array without changing the internal values, and checking the dtype) and isn't really meant for manipulating the contents/values of the array. Thus, I would keep `T[~np.isfinite(T)] = np.nan` in the subsequent `preprocess` functions, which DO change the actual values of the input array. Does that make sense?

Reply:
That makes sense :) Thank you for the explanation.